### PR TITLE
feat(compaction): support compact several files for overlap-strategy

### PR DESCRIPTION
**Pull request by Little-Wallace**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

* reduce default target_file_size_base to reduce write-amplification
* compare write-amplification score with tables in range rather than a single table, so that we can choose the best plan

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

**Comment by Little-Wallace** [comment]
main branch:
image
current PR:
image

File: src/meta/src/hummock/compaction/min_overlap_compaction_picker.rs [hunk 1]
```diff
@@ -29,137 +29,72 @@ pub struct MinOverlappingPicker {
...
     pub fn pick_tables(
         &self,
         select_tables: &[SstableInfo],
         target_tables: &[SstableInfo],
         level_handlers: &[LevelHandler],
     ) -> (Vec<SstableInfo>, Vec<SstableInfo>) {
         let mut scores = vec![];
-        for table in select_tables {
-            if level_handlers[self.level].is_pending_compact(&table.id) {
+        for left in 0..select_tables.len() {
```
Comment (Li0k) [inline src/meta/src/hummock/compaction/min_overlap_compaction_picker.rs:59]: IMHO, need some comments for this loop for readable

**Review by Li0k — APPROVED** [review]
Rest, LGTM !!! we can merge it after fix clippy. Thanks for this PR

File: src/meta/src/hummock/compaction/base_level_compaction_picker.rs [hunk 2]
```diff
@@ -132,21 +124,15 @@ impl CompactionPicker for LevelCompactionPicker {
             if write_amplification < all_level_amplification {
                 return None;
             }
+            // reverse because the ix of low sub-level is smaller.
+            input_levels.reverse();
```
Comment (Li0k) [inline src/meta/src/hummock/compaction/base_level_compaction_picker.rs:128]: A thought, do we need to add some functions to manage the logic corresponding to sub_level. Like the `reverse`, `sub_levels overlap type` and so on, the logic of other Levels is still a bit inconsistent

Comment (Little-Wallace) [reply src/meta/src/hummock/compaction/base_level_compaction_picker.rs:128]: Seems difficult. Maybe you can give another PR to fix it.

**Pull request merged**